src/Tests/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsArtifactPostProcessing.cs

```diff
@@ -0,0 +1,249 @@
+                                    Directory.GetFiles(testInstance.Path, "test3.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    "--collect:SampleDataCollector",
+                                    $"--test-adapter-path:{(merge ? Path.GetDirectoryName(s_dataCollectorDll) : Path.GetDirectoryName(s_dataCollectorNoMergeDll))}",
+                                    $"--settings:{runsettings}",
+                                    "--diag:" + testInstance.Path + "/logs/");
+
+            result.ExitCode.Should().Be(0);
+            AssertOutput(result.StdOut, merge);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ArtifactPostProcessing_VSTest_TestContainers(bool merge)
+        {
+            TestAsset testInstance = _testAssetsManager.CopyTestAsset("VSTestMultiProjectSolution", Guid.NewGuid().ToString())
+                .WithSource();
+
+            string runsettings = GetRunsetting(testInstance.Path);
+
+            new PublishCommand(Log, Path.Combine(testInstance.Path, "sln.sln")).Execute("/p:Configuration=Release").Should().Pass();
+
+            CommandResult result = new DotnetVSTestCommand(Log)
+              .WithWorkingDirectory(testInstance.Path)
+                                    .WithEnvironmentVariable(FeatureFlag.ARTIFACTS_POSTPROCESSING, "1")
+                                    .Execute(
+                                    Directory.GetFiles(testInstance.Path, "test1.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    Directory.GetFiles(testInstance.Path, "test2.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    Directory.GetFiles(testInstance.Path, "test3.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    "--collect:SampleDataCollector",
+                                    $"--testAdapterPath:{(merge ? Path.GetDirectoryName(s_dataCollectorDll) : Path.GetDirectoryName(s_dataCollectorNoMergeDll))}",
+                                    $"--settings:{runsettings}",
+                                    "--diag:" + testInstance.Path + "/logs/");
+
+            result.ExitCode.Should().Be(0);
+            AssertOutput(result.StdOut, merge);
+        }
+
+        private static void AssertOutput(string stdOut, bool merge)
+        {
+            List<string> output = new();
+            using StringReader reader = new(stdOut);
+            while (true)
+            {
+                string line = reader.ReadLine()?.Trim();
+                if (line is null) break;
+                output.Add(line);
+            }
+
+            if (merge)
+            {
+                Assert.Equal(string.Empty, output[output.Count - 3].Trim());
```

Comment:
I don't know if repo supports it but could you use the `^X` operator?

Reply:
I don't know...I'll check

Reply:
Works in local..let's see the CI